internal/compose/compose.go

```diff
@@ -275,49 +275,57 @@ func (p *Project) WaitForHealthy(opts CommandOptions) error {
 		return err
 	}
 
+	timeout := time.After(10 * time.Minute)
+	ticker := time.NewTicker(1 * time.Second)
+
 	containerIDs := strings.Split(strings.TrimSpace(b.String()), "\n")
 	for {
 		if signal.SIGINT() {
 			return errors.New("SIGINT: cancel waiting for policy assigned")
 		}
 
-		healthy := true
-
-		logger.Debugf("Wait for healthy containers: %s", strings.Join(containerIDs, ","))
-		descriptions, err := docker.InspectContainers(containerIDs...)
-		if err != nil {
-			return err
-		}
-
-		for _, containerDescription := range descriptions {
-			logger.Debugf("Container status: %s", containerDescription.String())
+		select {
+		case <-timeout:
+			return errors.New("time out waiting for healthy container")
+		case <-ticker.C:
+			healthy := true
 
-			// No healthcheck defined for service
-			if containerDescription.State.Status == "running" && containerDescription.State.Health == nil {
-				continue
+			logger.Debugf("Wait for healthy containers: %s", strings.Join(containerIDs, ","))
+			descriptions, err := docker.InspectContainers(containerIDs...)
+			if err != nil {
+				return err
 			}
-
-			// Service is up and running and it's healthy
-			if containerDescription.State.Status == "running" && containerDescription.State.Health.Status == "healthy" {
-				continue
+			for _, containerDescription := range descriptions {
+				logger.Debugf("Container status: %s", containerDescription.String())
+
+				// No healthcheck defined for service
+				if containerDescription.State.Status == "running" && containerDescription.State.Health == nil {
+					continue
+				}
+
+				// Service is up and running and it's healthy
+				if containerDescription.State.Status == "running" && containerDescription.State.Health.Status == "healthy" {
+					continue
+				}
+
+				// Container started and finished with exit code 0
+				if containerDescription.State.Status == "exited" && containerDescription.State.ExitCode == 0 {
+					continue
+				}
+
+				if containerDescription.State.Status == "exited" && containerDescription.State.ExitCode > 0 {
```

Comment:
The same line have been added in https://github.com/elastic/elastic-package/pull/688/commits/8563a1c8a7fec6375d82520d423ce33abf331b8f#diff-8019656c843609e3cb15ca57ab96dc41fdadae8215d0ff22d9b989f795aef4a4 so this PR will need to be rebased on top of `main` once https://github.com/elastic/elastic-package/pull/688 is merged

Reply:
removed